src/plugins/resolver/resolver.c

```diff
@@ -68,6 +80,16 @@ static void resolverInit (resolverHandle *p, const char *path)
 	p->tempfile = 0;
 
 	p->path = path;
+
+	{
+		pthread_mutexattr_t mutex_attr;
+
+		int result = pthread_mutexattr_init(&mutex_attr);
+		if(result == 0)
+			result = pthread_mutexattr_settype(&mutex_attr, PTHREAD_MUTEX_RECURSIVE);
+		if(result == 0)
+			result = pthread_mutex_init(getResolverMutex(), &mutex_attr);
```

Comment:
Oh yes. A 
   static int initialised = 0;
might do the trick. The calls are tacken from the POSIX documentation. So my guess is that the dynamic allocation should not fail except for the stack being limited in size for the pthread_mutex_t.